src/hotspot/cpu/aarch64/icBuffer_aarch64.cpp

```diff
@@ -52,9 +52,16 @@ void InlineCacheBuffer::assemble_ic_buffer_code(address code_begin, void* cached
   address start = __ pc();
   Label l;
   __ ldr(rscratch2, l);
-  __ far_jump(ExternalAddress(entry_point));
-  __ align(wordSize);
+  int jump_code_size = __ far_jump(ExternalAddress(entry_point));
+  // IC stub code size is not expected to vary depending on target address.
+  // We use NOPs to make the ldr+far_jump+int64 size equal to ic_stub_code_size.
+  for (int i = jump_code_size;
```

Comment:
Need more explanation for loop's boundaries arithmetic.
The comment includes `ldr` but you start from jump instruction size. Is it part of `3*instruction_size`? What are `3` instructions?

Reply:
I mean (ldr + int64 cached_value) = 4 + 8 = 12 = 3*instruction_size.
I change the code to make it clear now:
```C
  // We use NOPs to make the [ldr + far_jump + nops + int64] stub size equal to ic_stub_code_size.
  for (int size = NativeInstruction::instruction_size + jump_code_size + 8;
           size < ic_stub_code_size(); size += NativeInstruction::instruction_size) {
     __ nop();
  }
```